### PR TITLE
Checkstyle Improvements

### PR DESCRIPTION
**Pull request by AbdealiJK**

### CODE REVIEW

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/368/commits/fc9e9c9a085e79e2e38c6f2a7c5da6ec0a0f2f17

File: bears/java/CheckstyleBear.py [hunk 1]
```diff
@@ -25,6 +27,20 @@ class CheckstyleBear:
...
     LANGUAGES = "Java"
 
-    @staticmethod
-    def create_arguments(filename, file, config_file):
-        return '-jar', checkstyle_jar_file, '-c', google_checks, filename
+    known_configs = {
+        "google": os.path.join(DIR_PATH, 'google_checks.xml')}
+
+    def create_arguments(self, filename, file, config_file,
+                         checkstyle_configs: path="google"):
```
Comment (sils1297) [inline bears/java/CheckstyleBear.py]: not so sure, if I provide the value `google`, the path function will make an absolute path out of it

Comment (AbdealiJK) [inline bears/java/CheckstyleBear.py]: Tests pass, so I don't think so ?

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/368/commits/fe8c1ed70075c759e844a5dece83bc0dce36dc12 https://github.com/coala-analyzer/coala-bears/pull/368/commits/c25c0a3df43f7282afe1d3acf7889b5bf12ea709 https://github.com/coala-analyzer/coala-bears/pull/368/commits/73b99af422d9166143ee5fbbd94844c069fa313c

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala-bears/pull/368/commits/461017ad373dc6b00cec7a18fe8adcdd14598695

File: bears/java/CheckstyleBear.py (hunk 1 above)
Comment (sils1297) [inline bears/java/CheckstyleBear.py]: you don't test it with setting "google" explicitly

**Comment by AbdealiJK** [comment]
reack d47d714 a38e681

**Comment by AbdealiJK** [comment]
reack 6b896ddd746337f14c4a113ab1e3049ee4026b59 101b9e7451a473d042da7bb9449720e44f314c4f

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/368/commits/9aa4335fbba91b68206141b627712625bae8e4b3 https://github.com/coala-analyzer/coala-bears/pull/368/commits/841ec1d3c31bbd6d9c0f0c20cc706294bcfc3a60 https://github.com/coala-analyzer/coala-bears/pull/368/commits/cc1db31b106835ec27c1eb1f025c4d9ae89f063c https://github.com/coala-analyzer/coala-bears/pull/368/commits/b0c22754a34babfef09d98fc0c3b44b47f61abcf

**Comment by sils1297** [comment]
coverage is full

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**